web/client/epics/search.js

```diff
@@ -180,6 +186,32 @@ export const searchItemSelected = action$ =>
             return Rx.Observable.of(resultsPurge()).concat(itemSelectionStream, nestedServicesStream, searchTextStream);
         });
 
+/**
+ * Handles show GFI button click action.
+ */
+export const textSearchShowGFIEpic = (action$, store) =>
+    action$.ofType(TEXT_SEARCH_SHOW_GFI)
+        .switchMap(({item}) => {
+            const state = store.getState();
+            const typeName = item?.__SERVICE__?.options?.typeName;
+            const layerObj = typeName && getLayerFromName(state, typeName);
+            const bbox = item.bbox || item.properties.bbox || toBbox(item);
+            const coord = pointOnSurface(item).geometry.coordinates;
+            const latlng = { lng: coord[0], lat: coord[1] };
+
+            return !!coord &&
+                item?.__SERVICE__?.launchInfoPanel === 'single_layer' &&
```

Comment:
I see this selector (or a similar one) above in the PR. Consider if they can be refactored into a real selector

Reply:
@mbarto I'm sorry I don't understand what you mean. What selector are you reffering to?

Reply:
I see a pattern like this in two places:
```javascript
service.launchInfoPanel === 'single_layer' &&
                        !!service.openFeatureInfoButtonEnabled &&
                        (service.forceSearchLayerVisibility || !!find(this.props.layers, {name: service.options?.typeName})?.visibility)
```

Do you think you can factor it out as a function?